name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/ComdirectPDFExtractor.java

```diff
@@ -1809,249 +1822,191 @@ public List<Item> postProcessing(List<Item> items)
-                    {
-                        saleTransaction = (BuySellEntry) transactions.get(0).getSubject();
-                        taxesTransaction = (AccountTransaction) transactions.get(1).getSubject();
-                    }
-                    else if (transactions.get(1).getSubject() instanceof BuySellEntry
-                                    && transactions.get(0).getSubject() instanceof AccountTransaction)
-                    {
-                        saleTransaction = (BuySellEntry) transactions.get(1).getSubject();
-                        taxesTransaction = (AccountTransaction) transactions.get(0).getSubject();
-                    }
-
-                    // Check if not null and there is a sale transaction and a taxes transaction
-                    if (saleTransaction != null && taxesTransaction != null
-                                    && AccountTransaction.Type.TAXES.equals(taxesTransaction.getType())
-                                    && PortfolioTransaction.Type.SELL.equals(saleTransaction.getPortfolioTransaction().getType()))
-                    {
-                        // Subtract the tax from the taxes transaction from the total amount
-                        saleTransaction.setMonetaryAmount(saleTransaction.getPortfolioTransaction().getMonetaryAmount()
-                                        .subtract(taxesTransaction.getMonetaryAmount()));
-
-                        // Add taxes as tax unit
-                        saleTransaction.getPortfolioTransaction()
-                                        .addUnit(new Unit(Unit.Type.TAX, taxesTransaction.getMonetaryAmount()));
-
-                        // Combine at sources file (One or two files?)
-                        if (!saleTransaction.getSource().equals(taxesTransaction.getSource()))
-                            saleTransaction.setSource(saleTransaction.getSource() + "; " + taxesTransaction.getSource());
-
-                        // Combine at notes
-                        saleTransaction.setNote(concat(saleTransaction.getNote(), taxesTransaction.getNote()));
-
-                        // Add the taxes treatment to the hash table, which can be deleted
-                        TaxesTransactionToBeDelete.add(taxesTransaction);
-                    }
-                    else
-                    {
-                        // do nothing because no taxes transaction is present
-                    }
-                }
-            });
-        });
+        // Match transactions (sale + taxes treatment)
+        var saleTaxPairs = matchTransactionPair(saleTransactionList, taxesTreatmentList);
+
+        // Match transactions (dividend + taxes treatment)
+        var dividendTaxPairs = matchTransactionPair(dividendTransactionList, taxesTreatmentList);
+
+        // Fix missing currency conversion for sale tax transactions
+        fixMissingCurrencyConversionForSaleTaxesTransactions(saleTaxPairs);
+
+        // @formatter:off
```

Comment:
I understand that you need to switch off the formatter for text copied from the PDF documents - every space matters. I am not sure why it is needed for plain text comments - doesn't it just add two additional lines?

Reply:
I didn't even notice that anymore, but you're right. i reduced the comments.